backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/CPUPolicyUnit.java

```diff
@@ -47,14 +58,22 @@ public CPUPolicyUnit(PolicyUnit policyUnit,
                 continue;
             }
             Integer cores = SlaValidator.getEffectiveCpuCores(vds, context.getCluster().getCountThreadsAsCores());
-            if (cores != null && ((VmCpuCountHelper.isDynamicCpuTopologySet(vm) ?
-                    vm.getCurrentNumOfCpus(false) : vm.getNumOfCpus(false)) > cores)) {
-                messages.addMessage(vds.getId(), EngineMessage.VAR__DETAIL__NOT_ENOUGH_CORES.toString());
-                log.debug("Host '{}' has less cores ({}) than vm cores ({})",
-                        vds.getName(),
-                        cores,
-                        VmCpuCountHelper.getDynamicNumOfCpu(vm));
-                continue;
+            if (cores != null && vm.getCpuPinningPolicy() == CpuPinningPolicy.DEDICATED) {
+                cores = cores - takenCpus;
+            } else if (cores != null) {
+                cores = cores - takenCpus + 1; // adding back VDSM used CPU for a non-DEDICATED
```

Comment:
is it always a single cpu that is reserved for vdsm and the system processes? (@nyoxi)

Reply:
Yes, although returned as a list from VDSM.
Anyway, with current implementation we don't add this +1 CPU, we consider what is taken by dedicated.